From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Tue, 2 Jan 2024 20:54:01 +0100
Subject: [PATCH] Modded registries - Create class for block registry

* Creates a subclass of `Registry` for the block  registries: `BlockRegistry`.
* Changes the `BuiltInRegistries.BLOCK` registry instancs to have that type.
Does not add any behavior to that registry class.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/core/registries/BuiltInRegistries.java b/src/main/java/net/minecraft/core/registries/BuiltInRegistries.java
index ba7c778c8d29b529e28b3e48b606d899b33c2d3b..df491d7b512c401beff2014d0e7c14c5512fee99 100644
--- a/src/main/java/net/minecraft/core/registries/BuiltInRegistries.java
+++ b/src/main/java/net/minecraft/core/registries/BuiltInRegistries.java
@@ -125,7 +125,7 @@ public class BuiltInRegistries {
     });
     public static final DefaultedRegistry<Block> BLOCK = registerDefaultedWithIntrusiveHolders(Registries.BLOCK, "air", (registry) -> {
         return Blocks.AIR;
-    });
+    }, org.fiddlemc.fiddle.registry.BlockRegistry::new); // Fiddle - modded registries - class for block registry
     public static final Registry<Enchantment> ENCHANTMENT = registerSimple(Registries.ENCHANTMENT, (registry) -> {
         return Enchantments.BLOCK_FORTUNE;
     });
diff --git a/src/main/java/org/fiddlemc/fiddle/registry/BlockRegistry.java b/src/main/java/org/fiddlemc/fiddle/registry/BlockRegistry.java
new file mode 100644
index 0000000000000000000000000000000000000000..082c42a48b67a5d509b8553ddbc09991c8bd9051
--- /dev/null
+++ b/src/main/java/org/fiddlemc/fiddle/registry/BlockRegistry.java
@@ -0,0 +1,21 @@
+// Fiddle - modded registries - class for block registry
+
+package org.fiddlemc.fiddle.registry;
+
+import com.mojang.serialization.Lifecycle;
+import net.minecraft.core.DefaultedMappedRegistry;
+import net.minecraft.core.Registry;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.block.Block;
+
+/**
+ * An implementation of {@link Registry} specially for {@link BuiltInRegistries#BLOCK}.
+ */
+public final class BlockRegistry extends DefaultedMappedRegistry<Block> {
+
+    public BlockRegistry(String defaultId, ResourceKey<? extends Registry<Block>> key, Lifecycle lifecycle, boolean intrusive) {
+        super(defaultId, key, lifecycle, intrusive);
+    }
+
+}
